tests/unit/lms/services/lti_outcomes_test.py

```diff
@@ -300,23 +239,12 @@ def requests(self, patch):
         return patch("lms.services.lti_outcomes.requests")
 
     @classmethod
-    def element_text(cls, xml, path):
-        element = LTIOutcomesClient.find_element(xml, path)
-        if element is None:
-            return None
-        return element.text
+    def assert_sent_header_ok(cls):
+        """Check standard header fields of the request body."""
 
-    @classmethod
-    def check_header(cls, xml):
-        """Check standard header fields of an LTI Outcomes Management request body."""
-        assert (
-            cls.element_text(
-                xml,
-                [
-                    "imsx_POXHeader",
-                    "imsx_POXRequestHeaderInfo",
-                    "imsx_messageIdentifier",
-                ],
-            )
-            == "999999123"
-        )
+        body = cls.sent_body()
+
+        header = body["imsx_POXEnvelopeRequest"]["imsx_POXHeader"]
```

Comment:
Although it's not obvious from the code above, the previous code would have match "imsx_POXHeader" anywhere in the XML. We now specifically navigate from root.